src/python/shallow_water_visualization.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+from mpi4py import MPI
+from matplotlib import pyplot as plt, animation, colors
+import numpy as np
+
+from shallow_water_geometry_config import ShallowWaterGeometryConfig
+from shallow_water_geometry import ShallowWaterGeometry
+from shallow_water_model_config import ShallowWaterModelConfig
+from shallow_water_model import ShallowWaterModel
+from shallow_water_gt4py_config import ShallowWaterGT4PyConfig
+from shallow_water_state import ShallowWaterState
+
+def animate(n):
+    m.adv_nsteps(s, 1)
+    pc.set_array(s.h.data)
+
+class ColormapNormalize(colors.Normalize):
+    def __init__(self, vmin=None, vmax=None, v1=None, v2=None, clip=False):
+        self.v1 = v1
+        self.v2 = v2
+        super().__init__(vmin, vmax, clip)
+
+    def __call__(self, value, clip=None):
+        # I'm ignoring masked values and all kinds of edge cases to make a
+        # simple example...
+        # Note also that we must extrapolate beyond vmin/vmax
+        x, y = [self.vmin, self.v1, self.v2, self.vmax], [0, 0.05, 0.95, 1.]
+        return np.ma.masked_array(np.interp(value, x, y,
+                                            left=-np.inf, right=np.inf))
+
+    def inverse(self, value):
+        y, x = [self.vmin, self.v1, self.v2, self.vmax], [0, 0.1, 0.9, 1]
+
+        return np.interp(value, x, y, left=-np.inf, right=np.inf)
+
+comm = MPI.COMM_WORLD
+
+config_file = "../../parm/shallow_water.yml"
```

Comment:
You may want to generalize this in the same way you did with the main program by using command line arguments for consistency.

Reply:
I restructured this script to take in command line arguments similar to `shallow_water.py`.